app/src/test/java/com/monstarlab/features/auth/domain/AuthRepositoryTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.monstarlab.features.auth.domain
+
+import com.monstarlab.features.auth.AuthTokenMapper
+import com.monstarlab.features.auth.data.api.AuthApi
+import com.monstarlab.features.auth.data.api.dtos.TokenResponseDTO
+import com.monstarlab.features.auth.domain.models.AuthToken
+import io.mockk.*
+import junit.framework.TestCase.assertEquals
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+/**
+ * Created by aslam on 08,March,2023
+ */
+internal class AuthRepositoryTest {
+
+    private lateinit var mockAuthApi: AuthApi
+    private lateinit var mockAuthTokenMapper: AuthTokenMapper
+    private lateinit var authRepository: AuthRepository
+
+    @Before
+    fun setUp() {
+        mockAuthApi = mockk()
+        mockAuthTokenMapper = mockk()
+        authRepository = AuthRepository(mockAuthApi, mockAuthTokenMapper)
+    }
+
+    @After
+    fun tearDown() {
+        clearAllMocks()
+    }
+
+    @Test
+    fun testLoginReturnsExpectedAuthToken() = runBlocking {
+        // Arrange
+        val email = "aslam@test.com"
+        val password = "password"
+        val expectedTokenResponseDTO = TokenResponseDTO("token")
+        val expectedAuthToken = AuthToken("token")
+        coEvery { mockAuthApi.postLogin(email, password) } returns expectedTokenResponseDTO.also {
+            every {
+                mockAuthTokenMapper.map(it)
+            } returns expectedAuthToken
+        }
+
+        // Act
+        val actualAuthToken = authRepository.login(email, password)
+
+        // Assert
+        assertEquals(expectedAuthToken, actualAuthToken)
+        coVerify { mockAuthApi.postLogin(email, password) }.also {
+            verify {
+                mockAuthTokenMapper.map(expectedTokenResponseDTO)
+            }
+        }
+    }
```

Comment:
If you think about this test doesn't really test the real implementation
The repo has two functions:
1. call the API
2. Map the DTO to the domain class

Both those things are mocked so the only thing under the test here is that these mocks are actually called
The mapping mock can be removed for sure, there is no reason to mock it

Reply:
Regarding this test, I have mocked the API call and map function, and then performed an action on the unit that needs to be tested. After that, assertions and verification were used to check that the mock endpoint was called correctly. For better understanding, I have pushed a change. Please take a look.